packages/core/src/sync/index.ts

```diff
@@ -1274,9 +1276,11 @@ export async function* getLocalEventGenerator(params: {
       switch (filter.type) {
         case "log":
           if (isAddressFactory(filter.address)) {
-            const childAddresses = await params.syncStore.getChildAddresses({
-              factory: filter.address,
-            });
+            const childAddresses =
+              params.childAddressesCache.get(filter.address) ??
+              (await params.syncStore.getChildAddresses({
+                factory: filter.address,
+              }));
```

Comment:
When would the first expression evaluate to `undefined`? 

`sync-realtime` uses a design that pre-populates the cache. That way everywhere the cache is used it is guaranteed that the child addresses are already fetched.